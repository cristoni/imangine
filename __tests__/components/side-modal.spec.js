import React from 'react';
import SideModal from '../../components/side-modal';

test('SideModal renders its children when is open', () => {
  let wrapper = mount(<SideModal open={false}><div id="child" /></SideModal>);
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find("#child").length).toBe(0);

  wrapper = mount(<SideModal open={true}><div id="child" /></SideModal>);
  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find("#child").length).toBe(1);
});