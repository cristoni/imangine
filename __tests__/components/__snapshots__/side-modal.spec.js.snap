// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideModal renders its children when is open 1`] = `
<SideModal
  open={false}
>
  <div
    className="side-modal "
  />
</SideModal>
`;

exports[`SideModal renders its children when is open 2`] = `
<SideModal
  open={true}
>
  <div
    className="side-modal side-modal--open"
  >
    <Spring
      config={
        Object {
          "friction": 26,
          "tension": 170,
        }
      }
      force={false}
      from={
        Object {
          "opacity": 0.8,
          "transform": "translate3d(-100%,0,0)",
        }
      }
      immediate={false}
      inject={[Function]}
      native={false}
      reset={false}
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0%,0,0)",
        }
      }
    >
      <div
        className="side-modal__content shadow"
        onClick={[Function]}
        style={
          Object {
            "opacity": 0.8,
            "transform": "translate3d(-100%,0,0)",
          }
        }
      >
        <div
          className="side-modal__clear"
        >
          <i
            className="ml-2 material-icons"
          >
            clear
          </i>
        </div>
        <div
          id="child"
        />
      </div>
    </Spring>
  </div>
</SideModal>
`;
